### PR TITLE
Pin tool and package versions to make builds more repeatable

### PR DESCRIPTION
**Pull request by natemcmaster**
Part of aspnet/Universe#575 and aspnet/KoreBuild#239

This removes floating versions of dependencies. This includes the korebuild version and versions used by PackageReference's.

Changes:

 - (script generated) replace PackageLineup with MSBuild variables for dependencies versions

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<Project>
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+  <PropertyGroup Label="Package Versions">
+    <InternalAspNetCoreSdkPackageVersion>2.1.0-preview1-15549</InternalAspNetCoreSdkPackageVersion>
```
Comment (SteveSandersonMS) [inline build/dependencies.props]: @natemcmaster Not sure if this has been covered by email or in a brownbag or something, but could you briefly describe how these version numbers are going to get updated? Who does it, using what tool, and how often should we expect the updates to be applied?

Comment (natemcmaster) [reply build/dependencies.props]: It was in an email to the runtime team. The details are also posted here https://github.com/aspnet/KoreBuild/issues/239#issuecomment-340884986.

Short version:
`run.ps1 upgrade deps` will update the file to latest. Our CI will run it once a week on repos, but you're welcome to do it on your own and merge the results if you need newer dependencies.

Comment (SteveSandersonMS) [reply build/dependencies.props]: Sounds good - thanks for the info!

**Comment by natemcmaster** [comment]
Re-applied to dev in https://github.com/aspnet/JavaScriptServices/commit/2d98a1808ce49f9696c61099689e022d23774c75

**Pull request merged**